processor/transformprocessor/config.go

```diff
@@ -73,5 +87,9 @@ func (c *Config) Validate() error {
 		}
 	}
 
+	if c.FlattenData && !flatLogsFeatureGate.IsEnabled() {
```

Comment:
Would it be helpful to add a check to to see if `flatten_data` is being used when there are no `log_statements`? It's not an error, necessarily, but at least a warning log message to show that it's a no-op. Just a suggestion, I don't have a strong opinion here.

Reply:
I suppose the scenario where they could be confused is if they think `flatten_data` will apply to other data types. I considered naming the config options `flatten_logs` since it only applies to logs, but figured it might support other data types later. Not sure what to do here, also don't feel strongly about it.